src/realm/sync/subscriptions.cpp

```diff
@@ -151,6 +151,15 @@ size_t SubscriptionSet::size() const
     return m_sub_list.size();
 }
 
+Subscription SubscriptionSet::get_at_index(size_t index) const
+{
+    if (index >= m_sub_list.size()) {
+        return Subscription(this, Obj{});
```

Comment:
This should throw `std::out_of_range{}` instead of returning an empty Subscription.